consensus/core/src/authority_node.rs

```diff
@@ -42,6 +43,12 @@ pub enum ConsensusAuthority {
     WithTonic(AuthorityNode<TonicManager>),
 }
 
+/// A counter that keeps track of how many times the authority node has been booted while the binary
+/// or the component that is calling the `ConsensusAuthority` has been running. It's mostly useful to
+/// make decisions on whether amnesia recovery should run or not.
+
+pub type ConsensusAuthorityBootCounter = AtomicU64;
```

Comment:
Preferred this approach over initialising a static variable as this would not play out very well during unit testing with multiple nodes or even simtests which run as part of same binary and potential regressions could come up as we've seen in the past

Reply:
Agree. We may as well just pass a usize counter from the consensus manager to the consensus creation / start call. No need to expose this type and have it be atomic.

Reply:
The main reason why I preferred on creating this type is in order to have better control over the logic of the counter and the semantics. `ConsensusAuthority` will be responsible on when to increment the counter and interpret its value rather that the other way around. Having it atomic is not a big deal as it's just a non frequently updated one. I've considered this less error prone. Let me know if you feel otherwise .

Reply:
I like the idea of consensus manager playing the role of keeping track of the counter and when to increment it as it is the one in charge of starting it up and tearing it down.

Reply:
@arun-koshy sure , but it's kind of an internal logic/functionality to me as critical decisions need to be taken (start amnesia recovery or not) for which consensus manager might not bee accountable for. But I understand that you guys feel more natural place the consensus manager so I'll move it there.